Sources/Transport/Streams/ProgramStream/ProgramStream.swift

```diff
@@ -20,13 +22,27 @@ extension ProgramStream {
 }
 
 public enum SecurityLayer {
-    case none, tls
+    case none
+    case tls(TLSConfig)
+}
+
+extension SecurityLayer: Equatable {
+    public static func ==(lhs: SecurityLayer, rhs: SecurityLayer) -> Bool {
+        switch (lhs, rhs) {
+        case (.none, .none):
+            return true
+        case (.tls(_), .tls(_)):
+            return true
```

Comment:
@czechboy0 something about two different configs feels like equatable shouldn't return true.

Follow ups:

- do we need equatable on this?
- is there a situation where configs aren't equal but security layer should match?